src/cl65/main.c

```diff
@@ -541,7 +552,13 @@ static void AssembleFile (const char* File, unsigned ArgCount)
         ** to the file list of the linker. The name of the output
         ** file is that of the input file with ".s" replaced by ".o".
         */
-        char* ObjName = MakeFilename (File, ".o");
+        if (TmpFile) {
+            ObjName = MakeFilename (TmpFile, ".o");
+        }
```

Comment:
nitpicking: else should be in the same line as the previous closing brace.

I'll try to fix it myself before merge